services/system/AuthDelegate/src/AuthDelegate.cpp

```diff
@@ -40,6 +38,38 @@ namespace SystemService
       check(row.has_value(), "sender does not have an owning account");
    }
 
+   bool AuthDelegate::isAuthSys(psibase::AccountNumber              sender,
+                                std::vector<psibase::AccountNumber> authorizers)
+   {
+      auto record = db.open<AuthDelegateTable>().getIndex<0>().get(sender);
+      check(record.has_value(), "sender does not have an owning account");
```

Comment:
Since this appears in three different places, it could be factored out into a separate function.

Reply:
I miscounted. It's actually four places.

Reply:
done. removed other redundancies as well